### DISCUSSION TITLE
Real Time Meter Update

### DISCUSSION THREAD

**Discussion by ShivashishPachauri** [Akascape/TkDial#3] [General]
Hello Akascape, Great work for creating this library. I'm also interested to know if you have any feature to update the meter in real time based on some sensor value, for example Potentiometer. 
Does command parameter does the above mentioned thing? if yes, can you guide me for that?

**Comment by Akascape** [comment]
@ShivashishPachauri 
The command method is called when the needle is moved manually.
You can use `meter.set(value)` method to update the meter anytime. So when you get a sensor value, update the meter with `meter.set(sensor_value)`.
You can also disable the manual needle movement by configuring meter state to disabled.
Everything is given in the documentation.

**Comment by ShivashishPachauri** [comment]
import random
import tkinter as tk
from tkdial import Meter

class GUI():
        def ParsePacket():
            n = random.random()
            return n*100
        
          
if __name__ == "__main__":
      app = tk.Tk()
      meter1 = Meter(app,start=-180, end=180, start_angle=260, end_angle=-340,
               major_divisions=30,minor_divisions=5,fg="black", scale_color = "orange",needle_color="red",
               text_font="DS-Digital 10", text_color="orange")
      meter1.grid(padx=10, pady=10)        
      #meter1.set(0)
      while True :
          z=GUI.ParsePacket()
          meter1.set(z)
          print(z)
          app.mainloop()



I'm trying to run above code where I want meter to update in real time, but it does not update. For summary, **def ParsePacket()** function generates random numbers, and what I expect is to get these random number updates on Meter in real time. Im also attaching zip file of this code, in case above code have indentation errors. 

I tried putting **app.mainloop()** outside of while loop but still it didnt work. 

[RealTimeMeter.zip](https://github.com/Akascape/TkDial/files/11146499/RealTimeMeter.zip)

**Comment by Akascape** [comment]
@ShivashishPachauri You are using _while loop_ inside a _tkinter-mainloop_ which will never work for updating any widget value.
You should use threading in this case probably. Here is your working program:
```python
import random
import tkinter as tk
from tkdial import Meter
import threading
import time

class GUI():
    def ParsePacket():
        n = random.randint(-180, 180)
        return n
    
def update():
    while True:
        z = GUI.ParsePacket()
        meter1.set(z)
        print(z)
        time.sleep(1) # sleep for 1 second or else it will look weird

if __name__ == "__main__":
    app = tk.Tk()
    meter1 = Meter(app,start=-180, end=180, start_angle=260, end_angle=-340,
    major_divisions=30,minor_divisions=5,fg="black", scale_color = "orange",needle_color="red",
    text_font="DS-Digital 10", text_color="orange", state="disabled", scroll=False)
    meter1.grid(padx=10, pady=10)
    threading.Thread(target=update).start() # start this loop with threading
    app.mainloop()
```
